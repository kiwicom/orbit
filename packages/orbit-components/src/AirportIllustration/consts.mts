export const NAMES = [
  "BGYFastTrack",
  "BUDFastTrack",
  "MRSSmartPass",
  "NCEFastTrack",
  "PRGSmartPass",
  "VCESmartPass",
];
