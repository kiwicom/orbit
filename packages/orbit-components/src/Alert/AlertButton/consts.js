// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  INFO_SUBTLE: "infoSubtle",
  SUCCESS_SUBTLE: "successSubtle",
  WARNING_SUBTLE: "warningSubtle",
  CRITICAL_SUBTLE: "criticalSubtle",
  SECONDARY: "secondary",
};

export const TOKENS = {
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonFocus: "backgroundButtonFocus",
  colorTextButton: "colorTextButton",
  borderColorButtonFocus: "borderColorButtonFocus",
};
