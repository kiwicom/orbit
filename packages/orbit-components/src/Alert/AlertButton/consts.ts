export enum TYPE_OPTIONS {
  INFO = "info",
  SUCCESS = "success",
  WARNING = "warning",
  CRITICAL = "critical",
  INFO_SUBTLE = "infoSubtle",
  SUCCESS_SUBTLE = "successSubtle",
  WARNING_SUBTLE = "warningSubtle",
  CRITICAL_SUBTLE = "criticalSubtle",
  SECONDARY = "secondary",
}

export enum TOKENS {
  backgroundButton = "backgroundButton",
  backgroundButtonHover = "backgroundButtonHover",
  backgroundButtonActive = "backgroundButtonActive",
  backgroundButtonFocus = "backgroundButtonFocus",
  colorTextButton = "colorTextButton",
  borderColorButtonFocus = "borderColorButtonFocus",
}
