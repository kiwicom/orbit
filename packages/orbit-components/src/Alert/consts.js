// @flow
export const TYPE_OPTIONS = {
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
};

export const TOKENS = {
  backgroundAlert: "backgroundAlert",
  colorIconAlert: "colorIconAlert",
  colorTextLinkAlertHover: "colorTextLinkAlertHover",
  colorTextLinkAlertFocus: "colorTextLinkAlertFocus",
  colorBorderAlert: "colorBorderAlert",
  colorAccentBorder: "colorAccentBorder",
};

export const CLOSE_BUTTON_DATA_TEST = "AlertCloseButton";
