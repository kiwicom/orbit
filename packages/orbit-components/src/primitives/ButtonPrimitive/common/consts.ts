export enum SIZE_OPTIONS {
  SMALL = "small",
  NORMAL = "normal",
  LARGE = "large",
}

export enum TOKENS {
  heightButton = "heightButton",
  fontSizeButton = "fontSizeButton",
  paddingButton = "paddingButton",
  paddingButtonWithIcons = "paddingButtonWithIcons",
  paddingButtonWithLeftIcon = "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon = "paddingButtonWithRightIcon",
  spinnerWidth = "spinnerWidth",
  spinnerHeight = "spinnerHeight",
  marginRightIcon = "marginRightIcon",
}

export enum BUTTON_STATES {
  DEFAULT = "default",
  HOVER = "hover",
  ACTIVE = "active",
  FOCUS = "focus",
}
