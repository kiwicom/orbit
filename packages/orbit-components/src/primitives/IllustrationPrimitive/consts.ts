import { Size } from "./index.d";

type SizeKeys = "EXTRASMALL" | "SMALL" | "MEDIUM" | "LARGE" | "DISPLAY";

export const SIZE_OPTIONS: Record<SizeKeys, Size> = {
  EXTRASMALL: "extraSmall",
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large",
  DISPLAY: "display",
};

export const baseURL = "//images.kiwi.com";
