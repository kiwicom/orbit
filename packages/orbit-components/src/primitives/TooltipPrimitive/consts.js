// @flow

export const SIZE_OPTIONS = {
  SMALL: "small",
  MEDIUM: "medium",
};

// the default order on RTL needs to be changed
export const RTL_POSITIONS = {
  LEFT: "left",
  RIGHT: "right",
  TOP: "top",
  BOTTOM: "bottom",
};

export const POSITION_DIRECTIONS = {
  VERTICAL: "vertical",
  HORIZONTAL: "horizontal",
};

export const TOOLTIP_PADDING = 12;

export const TOOLTIP_ARROW_SIZE = 7;

export const TOOLTIP_TOTAL_PADDING: number = TOOLTIP_PADDING + TOOLTIP_ARROW_SIZE;
