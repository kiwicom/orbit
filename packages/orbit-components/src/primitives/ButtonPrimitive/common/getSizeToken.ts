import { SIZE_OPTIONS, TOKENS } from "./consts";
import type { Size } from "../types";
import type { Theme } from "../../../defaultTheme";

const getSizeToken = (
  size: Size,
  theme: Theme,
): {
  height: string;
  fontSize: string;
} => {
  const tokens = {
    [TOKENS.heightButton]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.formBoxLargeHeight,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.formBoxNormalHeight,
      [SIZE_OPTIONS.SMALL]: theme.orbit.formBoxSmallHeight,
    },
    [TOKENS.fontSizeButton]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.buttonLargeFontSize,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.buttonNormalFontSize,
      [SIZE_OPTIONS.SMALL]: theme.orbit.buttonSmallFontSize,
    },
    [TOKENS.spinnerWidth]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.iconMediumSize,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.iconMediumSize,
      [SIZE_OPTIONS.SMALL]: theme.orbit.iconSmallSize,
    },
    [TOKENS.spinnerHeight]: {
      [SIZE_OPTIONS.LARGE]: theme.orbit.iconMediumSize,
      [SIZE_OPTIONS.NORMAL]: theme.orbit.iconMediumSize,
      [SIZE_OPTIONS.SMALL]: theme.orbit.iconSmallSize,
    },
  };
  return {
    height: tokens[TOKENS.heightButton][size],
    fontSize: tokens[TOKENS.fontSizeButton][size],
  };
};

export default getSizeToken;
