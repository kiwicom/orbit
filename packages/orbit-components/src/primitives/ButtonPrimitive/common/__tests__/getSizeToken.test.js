// @flow
import getSizeToken from "../getSizeToken";
import { SIZE_OPTIONS } from "../consts";
import theme from "../../../../defaultTheme";

describe("ButtonPrimitive: getSizeToken function", () => {
  it("small size, should match snapshot", () => {
    const size = SIZE_OPTIONS.SMALL;
    expect(getSizeToken(size, theme)).toMatchSnapshot();
  });
  it("normal size, should match snapshot", () => {
    const size = SIZE_OPTIONS.NORMAL;
    expect(getSizeToken(size, theme)).toMatchSnapshot();
  });
  it("large size, should match snapshot", () => {
    const size = SIZE_OPTIONS.LARGE;
    expect(getSizeToken(size, theme)).toMatchSnapshot();
  });
});
