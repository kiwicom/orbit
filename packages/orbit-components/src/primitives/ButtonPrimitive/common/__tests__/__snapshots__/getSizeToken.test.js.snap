// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimitive: getSizeToken function large size, should match snapshot 1`] = `
Object {
  "fontSize": "16px",
  "height": "52px",
}
`;

exports[`ButtonPrimitive: getSizeToken function normal size, should match snapshot 1`] = `
Object {
  "fontSize": "14px",
  "height": "44px",
}
`;

exports[`ButtonPrimitive: getSizeToken function small size, should match snapshot 1`] = `
Object {
  "fontSize": "12px",
  "height": "32px",
}
`;
